src/storage/src/table/streaming_table/state_table.rs

```diff
@@ -612,6 +625,7 @@ impl<S: StateStore> StateTable<S> {
         &mut self,
         buffer: BTreeMap<Vec<u8>, RowOp>,
         epoch: u64,
+        watermark: Option<&ScalarImpl>,
```

Comment:
we might need to filter the data in the buffer with the watermark too, or the state store can not know which happens first.

Reply:
> we might need to filter the data in the buffer with the watermark too, or the state store can not know which happens first.

fixed